Modules/Common/src/TH1Ratio.cxx

```diff
@@ -0,0 +1,225 @@
+  mHistoDen->Sumw2();
+  Sumw2();
+
+  update();
+}
+
+void TH1Ratio::merge(MergeInterface* const other)
+{
+  if (!mHistoNum || !mHistoDen) {
+    return;
+  }
+
+  mHistoNum->Add(dynamic_cast<const TH1Ratio* const>(other)->getNum());
+  mHistoDen->Add(dynamic_cast<const TH1Ratio* const>(other)->getDen());
+  update();
+}
+
+void TH1Ratio::update()
+{
+  if (!mHistoNum || !mHistoDen) {
+    return;
+  }
+
+  const char* name = this->GetName();
+  const char* title = this->GetTitle();
+
+  TH1D::Reset();
+
+  SetNameTitle(name, title);
+  GetXaxis()->Set(mHistoNum->GetXaxis()->GetNbins(), mHistoNum->GetXaxis()->GetXmin(), mHistoNum->GetXaxis()->GetXmax());
+  SetBinsLength();
+
+  TH1D::Add(mHistoNum);
+
+  if (mUniformScaling) {
+    double entries = mHistoDen->GetBinContent(1);
+    double norm = (entries > 0) ? 1.0d / entries : 0;
+    Scale(norm);
+  } else {
+    Divide(mHistoDen);
+  }
+}
+
+void TH1Ratio::Reset(Option_t* option)
+{
+  if (mHistoNum) {
+    mHistoNum->Reset(option);
+  }
+
+  if (mHistoDen) {
+    mHistoDen->Reset(option);
+  }
+
+  TH1D::Reset(option);
+}
+
+void TH1Ratio::Copy(TObject& obj) const
+{
+  auto dest = dynamic_cast<TH1Ratio*>(&obj);
+  if (!dest) {
+    return;
+  }
+
+  getNum()->Copy(*(dest->getNum()));
+  getDen()->Copy(*(dest->getDen()));
+  TH1D::Copy(obj);
+
+  dest->update();
+}
+
+Bool_t TH1Ratio::Add(const TH1* h1, const TH1* h2, Double_t c1, Double_t c2)
+{
+  auto m1 = dynamic_cast<const TH1Ratio*>(h1);
+  if (!m1) {
+    return kFALSE;
+  }
+
+  auto m2 = dynamic_cast<const TH1Ratio*>(h2);
+  if (!m2) {
+    return kFALSE;
+  }
+
+  if (!getNum()->Add(m1->getNum(), m2->getNum(), c1, c2)) {
+    return kFALSE;
+  }
+  if (!getDen()->Add(m1->getDen(), m2->getDen(), c1, c2)) {
+    return kFALSE;
+  }
+
+  update();
+  return kTRUE;
+}
+
+Bool_t TH1Ratio::Add(const TH1* h1, Double_t c1)
+{
+  auto m1 = dynamic_cast<const TH1Ratio*>(h1);
+  if (!m1) {
+    return kFALSE;
+  }
+
+  if (!getNum()->Add(m1->getNum(), c1)) {
+    return kFALSE;
+  }
+  if (!getDen()->Add(m1->getDen(), c1)) {
+    return kFALSE;
+  }
+
+  update();
+  return kTRUE;
+}
+
+void TH1Ratio::SetBins(Int_t nx, Double_t xmin, Double_t xmax)
+{
+  getNum()->SetBins(nx, xmin, xmax);
+  getDen()->SetBins(nx, xmin, xmax);
+  TH1D::SetBins(nx, xmin, xmax);
+}
+
+void TH1Ratio::SetBins(Int_t nx, Double_t xmin, Double_t xmax,
+                       Int_t ny, Double_t ymin, Double_t ymax)
+{
+  ILOG(Debug, Devel) << "SetBins(Int_t nx, Double_t xmin, Double_t xmax, Int_t ny, Double_t ymin, Double_t ymax) not valid for MergeableTH2Ratio" << ENDM;
+}
+
+void TH1Ratio::SetBins(Int_t nx, Double_t xmin, Double_t xmax,
+                       Int_t ny, Double_t ymin, Double_t ymax,
+                       Int_t nz, Double_t zmin, Double_t zmax)
+{
+  ILOG(Debug, Devel) << "SetBins(Int_t nx, Double_t xmin, Double_t xmax, Int_t ny, Double_t ymin, Double_t ymax, Int_t nz, Double_t zmin, Double_t zmax) not valid for MergeableTH2Ratio" << ENDM;
+}
```

Comment:
I assume this comes from the ROOT's class. As far as I understand, we do not need these placeholder methods in our case, so perhaps you could remove them?